### ISSUE TITLE
Xen: Review the default for the new utilization parameters

### ISSUE THREAD

**Issue by diegoakechi** [ClusterLabs/resource-agents#1220]
Hi,

In a discussion with @krig about the [PR#1200](https://github.com/ClusterLabs/resource-agents/pull/1200/files), he realized that having the new utilization parameters set to _true_ can break the rolling upgrade for the users already using the Xen resource agent.

The problem is that utilization is a two part process: One sets utilization usage on resources, and utilization availability on nodes. However, the node utilization is not set automatically, the user has to set it.

The default utilization availability on nodes is not infinite, but zero. This means that a resource that has utilization set can never run anywhere.

So someone using the Xen resource today who upgrades will have the resource stopped and unable to run for unclear reasons.

This issue is to review this behavior and set the default to false. 

@oalbrigt and @vvidic could you give your thoughts on that?

Thanks.

**Comment by vvidic** [comment]
I saw that VirtualDomain agent has the default set to true, so I used the same here. So no problem for me if you want to change the utilization default to false.

**Comment by oalbrigt** [comment]
I think we should set it to false by default, and add info that utilization availability needs to be changed from 0 when you use this.

**Comment by vvidic** [comment]
Change from 0 or enable NodeUtilization agent, right?

**Comment by oalbrigt** [comment]
@vvidic: yeah.

**Comment by vvidic** [comment]
Want me to send a PR or you will fix directly?

**Comment by oalbrigt** [comment]
@vvidic A PR would be appreaciated.

**Comment by vvidic** [comment]
Ok, will try to send today.

**Comment by oalbrigt** [comment]
Thanks.

**Comment by vvidic** [comment]
Not sure what you mean, this should be a pacemaker thing?

**Comment by oalbrigt** [comment]
Oh, my bad.

**Comment by vvidic** [comment]
Yes, so the problem could happen on rolling upgrade if placement-strategy is set to something else than default. But I guess safe is better than sorry :)

**Comment by oalbrigt** [comment]
Yeah. Thanks for testing.

**Comment by krig** [comment]
Merged #1221. I also have an old PR for the VirtualDomain agent, I'd like to change the default to false there too. What do you guys think?

**Comment by oalbrigt** [comment]
Based on the placement-strategy comment I guess that makes sense.

**Comment by vvidic** [comment]
If utilization attributes stop being updated it could make them out of sync with the real VM state. Hard to tell if this can cause some real problems with resource placement later on.

**Comment by krig** [comment]
Perhaps ideally resource utilization updates should be automatically enabled only when the placement-strategy allows for it. But I don't know if that's doable.

**Comment by oalbrigt** [comment]
I guess we'd have to have it disabled by default, and a note in the metadata that it could cause issues if it gets out of sync with the VM state.
